### DISCUSSION TITLE
is PMD single class centric?

### DISCUSSION THREAD

**Discussion by saicharank** [pmd/pmd#4508] [Q&A]
Can i even write a custom rule which can detect unused public methods?

**Comment by jsotuyod** [comment]
PMD currently analyzes a single class in isolation, so no, there is no straight forward way of detecting unused public methods. There have been conversations about how we could extend this and enable multifile analysis, but so far these haven't materialized into any short-term implementation plans. Help and thoughts on this are more than welcomed.

**Reply by NilsBaumgartner1994** [reply]
Ah dang, thats what i was hoping for. I am hoping to detect data clumps, therefore i would also need to know public used methods and when the global check starts and finishes.
